compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -139,8 +139,8 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
         let mut var_data = self.construct_var_data();
 
         // Deduplicating constraints is shown to have a positive perf impact.
-        self.data.constraints.sort_by_key(|(constraint, _)| *constraint);
```

Comment:
Would you mind using an [UnordSet](https://doc.rust-lang.org/stable/nightly-rustc/rustc_data_structures/unord/struct.UnordSet.html) here? It is a drop-in replacement in this case but guards against accidentally relying on its iteration order.

Reply:
doesn't that have the same insertion issues as https://github.com/rust-lang/rust/pull/118824#issuecomment-1857798464 (expensive to insert)

Reply:
or do you mean for the dedup set? sure, that doesn't matter

Reply:
Yeah, I meant the `seen` set. The comment ended up in an ambiguous place 🙂